Packs/Base/Scripts/CommonServerPython/CommonServerPython.py

```diff
@@ -10986,15 +11009,45 @@ def events_error_handler(res):
 
     zipped_data = gzip.compress(data.encode('utf-8'))  # type: ignore[AttributeError,attr-defined]
     client = BaseClient(base_url=xsiam_url)
+    data_chunks = split_xsiam_events_to_chunks(data, sys.getsizeof(zipped_data))
+    for data_chunk in data_chunks:
+        amount_of_events = len(data_chunk)
+        data_chunk = '\n'.join(data_chunk)
+        zipped_data = gzip.compress(data_chunk.encode('utf-8'))  # type: ignore[AttributeError,attr-defined]
+        send_to_xsiam_with_retries(client, events_error_handler, header_msg, headers, num_of_attempts, xsiam_url,
+                                   zipped_data)
+        demisto.updateModuleHealth({'eventsPulled': amount_of_events})
+
+
+def send_to_xsiam_with_retries(client, events_error_handler, header_msg, headers, num_of_attempts, xsiam_url,
+                               zipped_data):
+    """
+    Sends zipped data to XSIAM with retries.
 
+    :type client: ``base client object``
+    :param client: An instance of the client class.
+    :type events_error_handler: ``function``
+    :param events_error_handler: A function that handles events errors.
+    :type header_msg: ``str``
+    :param header_msg: The header message.
+    :type headers: ``dict``
+    :param headers: A dictionary containing headers for the request.
+    :type num_of_attempts: ``int``
+    :param num_of_attempts: The number of attempts to send the data to XSIAM.
+    :type xsiam_url: ``str``
+    :param xsiam_url: The URL of the XSIAM server.
+    :type zipped_data: ``bytes``
+    :param zipped_data: The zipped data to send.
+
+    :return: Bool or DemistoException
```

Comment:
`return` is the same as `rtype`?

Reply:
for some reason i dont understand yet, it parses the docstring successfully with rtype and fails with return. 
In validations when it runs the commonServerDocs.sh script.